### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by redwing555**
This school library project's milestone consists of  the following requirements : 
- Creating a class `Classroom` with the following:

    1. `@Label` instance variable is initialized in the constructor.
    2. Setter and getter for `@Label` .

- Creating the has-many/belongs-to relationship between Classroom and Student with the following:

    1. Creating the has-many side (a `classroom` has many students).
    2. Creating the belongs-to side (a `student` belongs to a classroom).
    3. Making sure that when adding a `student` to a `classroom` it also sets the classroom for the student.
    4. Making sure that when setting the` classroom` for a `student` it also adds it to the classrooms' students.

 - Creating a class `Book` with the following:

    1. `@title` and `@author` instance variables are initialized in the constructor.
    2. Setters and getters for instance variables.

 - Creating a class `Rental` with the following:

    1. `@Date` instance variable is initialized in the constructor.
    2. Setter and getter for `@Date`.

 - Creating the many-to-many (also has-many-through) relationship between `Person` and `Book` using the class `Rental` :

      1. Creating the has-many side of `Book` and `Rental` (a book has many rentals).
      2. Creating the belongs-to side of `Rental` and `Book` (a rental belongs to a book).
      3. Creating the has-many side of `Person` and `Rental` (a person has many rentals).
      4. Creating the belongs-to side of `Rental` and `Person` (a rental belongs to a person).
      5. Modifying the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
   end
+
+  def add_classroom(classroom)
+    if @classroom.include?(classroom)
+      puts 'Student already enrolled in this classroom'
+    else
+      classroom.add_student(self)
+      @classroom.push(classroom)
+    end
+  end
```
Comment (D4nielJ) [inline student.rb]: - [ ] As indicated in the requirements, the relation between student and classroom is a student belongs to... In this case, we want to make sure that the instance class points only to the classroom where the student belongs. So I think we should not use an array here, instead just reference the one classroom.

>Creating the has-many side (a classroom has many students).
Creating the belongs-to side (a student belongs to a classroom).
Making sure that when adding a student to a classroom it also sets the classroom for the student.

**Review by D4nielJ — CHANGES_REQUESTED** [review]
## Hello @redwing555, 

You're doing great! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights :fire: :rocket: 
- Great job using accessor to config getters and setters!
- Most of the relationships are very well structured
- Good looking and clean code! :sparkles: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

## Happy coding and stay awesome! 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lameck1 — APPROVED** [review]
Hi, @redwing555,

You've done well working on the previous review concerns 👏🏼.
The tasks for this milestone are therefore successfully implemented 💯.
All the best with the next milestone.

You are free to merge this branch :shipit:

Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by redwing555** [comment]
@Lameck1  @D4nielJ  thank you very much for the review, cheers and happy coding 🥂

**Pull request merged**